imageio/plugins/ffmpeg.py

```diff
@@ -66,22 +82,16 @@ def get_exe():
     if exe:  # pragma: no cover
         return exe
 
-    # Check if ffmpeg is in PATH
-    try:
-        with open(os.devnull, "w") as null:
-            sp.check_call(["ffmpeg", "-version"], stdout=null,
-                          stderr=sp.STDOUT)
-            return "ffmpeg"
-    # ValueError is raised on failure on OS X through Python 2.7.11
-    # https://bugs.python.org/issue26083
-    except (OSError, ValueError, sp.CalledProcessError):
-        pass
-    
     plat = get_platform()
+
+    # To give the executable provided by imageio highest priority,
+    # generate a list of executables and allow to add executables 
+    # to the beginning or the end according to priority.
+    # The initial string "ffmpeg" is a fallback in case nothing is found.
+    exes = ["ffmpeg"]
     
     # Check if ffmpeg is installed in Python environment
     # (e.g. via conda install ffmpeg -c conda-forge)
-    exe = None
```

Comment:
I'm not yet happy with this part :) E.g. exes is always nonempty, but at the end we test whether it is.

I think it would be better to forget about that list and just try each option in order of priority, returning the value if it was found to be correct.

* If `IMAGEIO_FFMPEG_EXE` is set, use that (even without testing whether it is a valid path).
* Try our own version using `get_remote_file()`. (if `NeedDownloadError` is raised, just continue to next step)
* Try from conda package
* Try `ffmpeg` command
* Nothing found, raise `NeedDownloadError` with nice message.